src/controller/proxy/local.go

```diff
@@ -0,0 +1,199 @@
+//  Copyright Project Harbor Authors
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+
+package proxy
+
+import (
+	"context"
+	"errors"
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/manifest/manifestlist"
+	comHttpAuth "github.com/goharbor/harbor/src/common/http/modifier/auth"
+	"github.com/goharbor/harbor/src/controller/artifact"
+	"github.com/goharbor/harbor/src/core/config"
+	"github.com/goharbor/harbor/src/lib"
+	"github.com/goharbor/harbor/src/lib/log"
+	"github.com/goharbor/harbor/src/pkg/registry"
+	"io"
+	"time"
+)
+
+var errManifestNotFound = errors.New("manifest list doesn't contain any pushed manifest")
+
+// localInterface defines operations related to local repo under proxy mode
+type localInterface interface {
+	// BlobExist check if the blob exist in local repo
+	BlobExist(ctx context.Context, art lib.ArtifactInfo) (bool, error)
+	// Manifest check if the manifest exist in local repo
+	ManifestExist(ctx context.Context, art lib.ArtifactInfo) (bool, error)
```

Comment:
It should not return error and the error should not be checked.

Reply:
Updated